src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
```

Comment:
PullProofAI

There is a potential performance issue in the `JwtStrategy` class. Accessing the Redis service synchronously for each validation request can introduce latency and impact performance, especially under high load.

#### Suggestions
Consider optimizing the usage of the Redis service by implementing a caching mechanism or reducing unnecessary calls to improve performance. One approach could be to cache JWT tokens locally after retrieval to avoid repeated Redis queries.

```diff
+ // Consider implementing a caching mechanism for JWT tokens to reduce Redis calls
+ // Example: Cache JWT tokens locally after retrieval to avoid repeated Redis queries
```


<!-- This is an auto-generated comment by PullProofAI -->